### PR TITLE
Avoid conda-4.6.1 in staged-recipes builds

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This PR hacks the build scripts for all three CI providers so avoid `conda-4.6.1`, see https://github.com/conda/conda/issues/8150.

### CODE REVIEW

**Comment by duncanmmacleod** [comment]
@conda-forge/core, is this a reasonable short-term workaround to the current staged-recipes PR failures?

**Comment by mariusvniekerk** [comment]
Why are we still using ci setup 1.x?

**Comment by duncanmmacleod** [comment]
@mariusvniekerk, I'm not sure (the question probably wasn't addressed at me), but it does beg another question of whether this change (avoid conda 4.6.1) is better suited to go upstream into conda-forge-ci-setup?

**Comment by beckermr** [comment]
@mariusvniekerk I think the ci setup at 1.* is a bug.

**Comment by scopatz** [comment]
Can we fix the ci setup 1.* issue?

**Comment by duncanmmacleod** [comment]
@scopatz, can you suggest a patch, and I'll add it to this PR?

**Comment by scopatz** [comment]
I am not sure what needs to be done myself, here.

File: .appveyor.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ install:
     - cmd: appveyor-retry conda.exe update --yes --quiet conda
 
 
-    - cmd: appveyor-retry conda.exe install --yes --quiet conda-forge-pinning conda-forge-ci-setup=1.* networkx conda-build>=3.16
+    - cmd: appveyor-retry conda.exe install --yes --quiet "conda!=4.6.1" conda-forge-pinning conda-forge-ci-setup=2.* networkx conda-build>=3.16
```
Comment (scopatz) [inline .appveyor.yml]: Can you remove the "conda!=4.6.1" bits until we try it with the proper version of ci-setup?

Comment (duncanmmacleod) [reply .appveyor.yml]: @scopatz, the current Travis CI builds for staged-recipes are failing with conda-4.6.1, even with `conda-forge-ci-setup=2.*`.

Comment (scopatz) [reply .appveyor.yml]: Hmm do you have a link to a build that shows that?  I reallt just don't like pinning in this way, because we have to undo it later.

Comment (duncanmmacleod) [reply .appveyor.yml]: @scopatz, https://travis-ci.org/conda-forge/staged-recipes/builds/485851196

Comment (scopatz) [reply .appveyor.yml]: No avoiding it then, let's try it!

**Comment by scopatz** [comment]
Thanks @duncanmmacleod

**Pull request merged**